### DISCUSSION TITLE
cake 1.0.0 with GitVersion fails on Linux and or Windows

### DISCUSSION THREAD

**Discussion by PulsarFX** [cake-build/cake#3154] [Q&A] [answered]
### What You Are Seeing?

With `#tool nuget:?package=GitVersion.Tool&version=5.6.6`  in my build.cake I get this output:
> Error: One or more errors occurred. (GitVersion: Could not locate executable.)

This fails on Linux and Windows with the same message. 

With  `#tool nuget:?package=GitVersion.CommandLine&version=5.6.6`  I get this output:

It works on Windows, but fails on Linux:
>run-detectors: unable to find an interpreter for /home/.../myagent/_work/4/s/tools/GitVersion.CommandLine.5.6.6/tools/gitversion.exe
> Error: One or more errors occurred. (GitVersion: Process returned an error (exit code 2).)

### What is Expected?
GitVersion works with one of the GitVersion Tools

### What version of Cake are you using?
cake 1.0.0

### Are you running on a 32 or 64 bit system?
64bit

### How Did You Get This To Happen? (Steps to Reproduce)
demo cake file
```
#tool "nuget:?package=GitVersion.CommandLine&version=5.6.6"

var version = GitVersion();


Task("Build")
        .Does(() =>
{
        Information($"Done");
});
```

### Output Log
cake 1.0.0 fails with GitVersion.CommandLine tool 
https://gist.github.com/maulwuff/c524c9e7fe0775f740682003050b183f

cake 1.0.0 fails with GitVersion.Tool tool 
https://gist.github.com/maulwuff/d88f42db02324030a3f24ec75ac15ca2

```
$ dotnet --info
.NET SDK (gemäß "global.json"):
 Version:   5.0.103
 Commit:    9effbc8ad5

Laufzeitumgebung:
 OS Name:     debian
 OS Version:  9
 OS Platform: Linux
 RID:         debian.9-x64
 Base Path:   /usr/share/dotnet/sdk/5.0.103/

Host (useful for support):
  Version: 5.0.3
  Commit:  eae88cc11b

.NET SDKs installed:
  2.1.802 [/usr/share/dotnet/sdk]
  3.0.101 [/usr/share/dotnet/sdk]
  5.0.103 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.14 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.14 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.14 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

**Comment by devlead** [comment, marked as answer]
GitVersion tool is a .NET tool so needs to be installed through the .NET CLI, fortunately there's a Cake module called  `Cake.DotNetTool.Module` that takes care of that for you, example usage:


```cake
// Install modules (if you are using Cake v1.1 or higher, you *don't* need this line)
#module nuget:?package=Cake.DotNetTool.Module&version=0.4.0

// Install .NET Core Global tools.
#tool "dotnet:?package=GitVersion.Tool&version=5.1.2"
```

**Reply by PulsarFX** [reply]
Thanks, it works with your example. I used the latest versions:
```
#module nuget:?package=Cake.DotNetTool.Module&version=0.4.0
#tool dotnet:?package=GitVersion.Tool&version=5.6.6
```

**Reply by augustoproiete** [reply]
Just a quick note for others that arrive here from search engines: As of [Cake v1.1 the DotNetTool module is included with Cake](https://cakebuild.net/blog/2021/03/cake-v1.1.0-released), and there's no need to install the `Cake.DotNetTool.Module` anymore.

- [Documentation on the .NET Tool Provider](https://cakebuild.net/docs/writing-builds/preprocessor-directives/tool/dotnet-provider)
